src/app/pages/deposits/deposits.component.ts

```diff
@@ -0,0 +1,74 @@
+import { Deposit } from 'src/app/models/Deposit';
+import { Component, OnInit, OnDestroy } from '@angular/core';
+import { ToastrService } from 'ngx-toastr';
+import { NgxSpinnerService } from 'ngx-spinner';
+import { Subscription } from 'rxjs';
+
+import { DepositsService } from 'src/app/services/deposits/deposits.service';
+import { removeSubscription } from 'src/app/helpers/unsubscribe';
+
+@Component({
+  selector: 'app-deposits',
+  templateUrl: './deposits.component.html',
+  styleUrls: ['./deposits.component.scss']
+})
+export class DepositsComponent implements OnInit, OnDestroy {
+  subscribe: Subscription[] = [];
+  headElements = ['Reference', 'Date', 'Amount'];
+  transactions: Deposit[] = [];
+  startDateVal: string = '';
+  endDateVal: string = '';
+  newResults: any[];
+
+  constructor(
+    private depositsService: DepositsService,
+    private spinner: NgxSpinnerService,
+    private toastrService: ToastrService
+  ) {}
+
+  onFilter() {
+    const filteredTransactions = [];
+    this.transactions.filter(deposit => {
+      const date = new Date(deposit.created_at);
+      const fromDate = new Date(this.startDateVal);
+      const toDate = new Date(this.endDateVal);
+
+      if (fromDate && toDate) {
+        if (date >= fromDate && date <= toDate) {
+          filteredTransactions.push(deposit);
+        }
+      }
+    });
+    this.transactions = filteredTransactions;
+    this.newResults = filteredTransactions;
+  }
+
+  clear() {
+    this.newResults = null;
+    this.setTransactions();
+  }
+
+  ngOnInit() {
+    this.setTransactions();
+  }
+
+  setTransactions() {
+    this.spinner.show();
+    this.subscribe.push(
+      this.depositsService.getDeposits().subscribe(
+        response => {
+          this.transactions = response.results;
+          this.spinner.hide();
+        },
+        error => {
```

Comment:
Nice work @Genza999 , theres a tiny issue here. Not all errors are coming in are of the shape 

```
error.errors
```

take a look at the error I'm getting from my end

![image](https://user-images.githubusercontent.com/12128153/63936717-3faa1a00-ca69-11e9-9a66-1c068a0359bf.png)

This error is actually a string, I think its a problem from the backend. what you can do is this

```js
this.toastrService.error(error.errors || "Something went wrong fetching your deposits");
```

in this case when there is no error comming from the object then it will fallback to the text.

Otherwise, users will see this empty error toast and not know whatsup

![image](https://user-images.githubusercontent.com/12128153/63936845-9adc0c80-ca69-11e9-9f59-9b253d2c64a3.png)

Reply:
Thanks @Tevinthuku , I have implemented this!